lib/profile_finder.js

```diff
@@ -16,6 +16,14 @@ function locateUserDirectory(platform) {
       break;
     case 'linux':
       userDir = path.join(process.env.HOME, '/.mozilla/firefox');
+      // fix: https://github.com/saadtazi/firefox-profile-js/issues/136
+      // check if directory is present, otherwise use ubuntu path: ~/snap/firefox/common/.mozilla/firefox/
+      if (!fs.existsSync(userDir)) {
```

Comment:
I haven't tested my code yet, but it was along the same lines:

```js
snapDir = ...
dotDir = ...

xdgConfigDir = process.env.XDG_CONFIG_HOME || path.join(process.env.HOME, ".config")
xdgDir = path.join(xdgConfigDir, "mozilla/firefox")
if (fs.existsSync(dotDir)) { userDir = dotDir }
else if (fs.existsSync(snapDir)) { userDir = snapDir }
else { userDir = xdgDir }
```

Reply:
Ah, I wasn't sure what the XDG_CONFIG_HOME was used for, thank you!

Reply:
Pushed some changes, it's now closer to the logic you proposed.